subworkflows/local/utils_nfcore_detaxizer_pipeline/main.nf

```diff
@@ -182,26 +182,30 @@ def genomeExistsError() {
 // Generate methods description for MultiQC
 //
 def toolCitationText() {
-    // TODO nf-core: Optionally add in-text citation tools to this list.
-    // Can use ternary operators to dynamically construct based conditions, e.g. params["run_xyz"] ? "Tool (Foo et al. 2023)" : "",
-    // Uncomment function in methodsDescriptionText to render in MultiQC report
+
     def citation_text = [
             "Tools used in the workflow included:",
             "FastQC (Andrews 2010),",
             "MultiQC (Ewels et al. 2016)",
+            "Kraken2 (Wood et al. 2019),",
+            !params["skip_blastn"] ? "BLAST (Altschul et al. 1990)," : "",
+            !params["skip_blastn"] | params["enable_filter"] ? "seqkit (Shen et al. 2016)," : "",
+            "fastp (Chen et al. 2018)",
```

Comment:
you could roughly keep the sequence of the tools, i.e. to put fastp right after FastQC or such.